src/v/cluster/rm_stm.h

```diff
@@ -419,6 +419,12 @@ class rm_stm final : public persisted_stm {
         absl::flat_hash_map<model::producer_identity, expiration_info>
           expiration;
 
+        // We use this map to store pid for idempotent request, and time when we
+        // replicate batch with seq_number for this pid. We need this info to
+        // delete old pids
+        absl::btree_multimap<time_point_type, model::producer_identity>
```

Comment:
this field isn't populated from the log via `apply` or `snapshot` so it shouldn't be part of the log_state. mem_state?